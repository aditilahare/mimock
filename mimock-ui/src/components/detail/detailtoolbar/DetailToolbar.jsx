import React from 'react';
import {
	DetailToolbarContainer,
	MiniBtnSpan,
	BackButton,
} from './DetailToolbar.style';
import { constants } from './constants';
import { FaArrowCircleLeft } from 'react-icons/fa';
import PropTypes from 'prop-types';

function DetailToolbar() {
	return (
		<DetailToolbarContainer data-testid='detail-toolbar-section'>
			<BackButton to={`/mimock-ui/mocks`} dataTestid='back-btn'>
				<MiniBtnSpan>
					<FaArrowCircleLeft /> {constants.label.backBtn}
				</MiniBtnSpan>
			</BackButton>
		</DetailToolbarContainer>
	);
}

DetailToolbar.propTypes = {
	handleBackBtnClick: PropTypes.func.isRequired,
};

export default DetailToolbar;
