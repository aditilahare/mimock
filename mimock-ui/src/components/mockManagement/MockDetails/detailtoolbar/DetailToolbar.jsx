import React from 'react';
import PropTypes from 'prop-types';
import {
	DetailToolbarContainer,
	MiniBtnSpan,
	BackButton,
	BaseButton,
	OperationsContainer,
	WarningMessage,
	ArchiveIcon,
	UnArchiveIcon,
	DeleteIcon,
	EditIcon,
	BackIcon,
	ReportProblemIcon,
} from './DetailToolbar.style';
import { constants } from './constants';

function DetailToolbar({
	hideDetailActionsToolbar,
	status,
	performArchiveMockOperation,
	performUnarchiveMockOperation,
	performDeleteMockOperation,
	performEditMockOperation,
}) {
	// #region Defaults
	const { testIds, ids, label } = constants;
	// #endregion

	return (
		<DetailToolbarContainer data-testid={testIds.detailToolbarContainer}>
			<BackButton
				to={`/mimock-ui/mocks`}
				data-testid={testIds.backBtn}
				id={ids.backBtn}
			>
				<MiniBtnSpan>
					<BackIcon /> {label.backBtn}
				</MiniBtnSpan>
			</BackButton>
			<If condition={!hideDetailActionsToolbar}>
				<OperationsContainer data-testid={testIds.operationsContainer}>
					<Choose>
						<When condition={status == 'NONE'}>
							<BaseButton
								data-testid={testIds.archiveBtn}
								id={ids.archiveBtn}
								onClick={performArchiveMockOperation}
							>
								<MiniBtnSpan>
									<ArchiveIcon /> {label.archiveBtn}
								</MiniBtnSpan>
							</BaseButton>
							<BaseButton
								data-testid={testIds.deleteBtn}
								id={ids.deleteBtn}
								onClick={performDeleteMockOperation}
							>
								<MiniBtnSpan>
									<DeleteIcon /> {label.deleteBtn}
								</MiniBtnSpan>
							</BaseButton>
							<BaseButton
								data-testid={testIds.editBtn}
								id={ids.editBtn}
								onClick={performEditMockOperation}
							>
								<MiniBtnSpan>
									<EditIcon /> {label.editBtn}
								</MiniBtnSpan>
							</BaseButton>
						</When>
						<When condition={status == 'ARCHIVED'}>
							<BaseButton
								data-testid={testIds.unarchiveBtn}
								id={ids.unarchiveBtn}
								onClick={performUnarchiveMockOperation}
							>
								<MiniBtnSpan>
									<UnArchiveIcon /> {label.unarchiveBtn}
								</MiniBtnSpan>
							</BaseButton>
						</When>
						<When condition={status == 'DELETED'}>
							<WarningMessage
								data-testid={testIds.warningMessage}
								id={ids.warningMessage}
							>
								<ReportProblemIcon /> {label.warningMessage}
							</WarningMessage>
						</When>
					</Choose>
				</OperationsContainer>
			</If>
		</DetailToolbarContainer>
	);
}

DetailToolbar.propTypes = {
	hideDetailActionsToolbar: PropTypes.bool.isRequired,
	status: PropTypes.string.isRequired,
	performArchiveMockOperation: PropTypes.func.isRequired,
	performUnarchiveMockOperation: PropTypes.func.isRequired,
	performDeleteMockOperation: PropTypes.func.isRequired,
	performEditMockOperation: PropTypes.func.isRequired,
};

export default DetailToolbar;
