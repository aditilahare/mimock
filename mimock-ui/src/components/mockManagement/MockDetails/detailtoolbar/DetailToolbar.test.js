import React from 'react';
import { render } from '@testing-library/react';
import DetailToolbar from './DetailToolbar';
import { BrowserRouter } from 'react-router-dom';

describe('DetailToolbar', () => {
	it('should render detail toolbar component', async () => {
		const tree = await render(<DetailToolbar />, { wrapper: BrowserRouter });

		const { container, getByTestId } = tree;

		expect(getByTestId('detail-toolbar-section')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
