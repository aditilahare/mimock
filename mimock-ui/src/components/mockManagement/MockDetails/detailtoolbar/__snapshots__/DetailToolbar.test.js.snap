// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailToolbar should render detail toolbar component 1`] = `
<div>
  <div
    class="flex flex-row justify-between align-middle w-full mt-4 mb-2"
    data-testid="detail-toolbar-container"
  >
    <a
      class="sc-bdvvtL px-2 py-2 mx-1 hover:bg-gray-100"
      data-testid="back-btn"
      href="/mimock-ui/mocks"
      id="back-btn"
    >
      <span
        class="inline-flex items-center justify-between gap-2 text-lg text-gray-600"
      >
        <svg
          class="text-2xl text-gray-600"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
          />
        </svg>
         
        Back
      </span>
    </a>
    <div
      class="inline-flex items-center justify-between"
      data-testid="operations-container"
    />
  </div>
</div>
`;
